builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -1626,7 +1704,7 @@ func (vm *virtualMachine) setupVirtualMachine(c *govmomi.Client) error {
 	for _, network := range vm.networkInterfaces {
 		// network device
 		var networkDeviceType string
-		if vm.template == "" {
+		if vm.clone == nil || vm.clone.source == "" {
```

Comment:
Totally unrelated, but shouldn't we be setting network type based on a schema value, and why is it forced to e1000 when it is a clone.  Nothing that needs to be fixed in this PR, sorry for the rant.